### PR TITLE
Added dispatchJSEventOnElement() method to be able to execute Events

### PR DESCRIPTION
**Pull request by rashidsaf**
Added dispatchJSEventOnElement() method to be able to execute Events

### CODE REVIEW

File: src/Selenium2Driver.php [hunk 1]
```diff
@@ -278,6 +278,24 @@ private function executeJsOnElement(Element $element, $script, $sync = true)
...
+     */
+    public function dispatchJSEventOnElement($event, $xpath)
+    {
+        $script = <<<JS
+(function (element) {
+var event = document.createEvent("HTMLEvents");
+event.initEvent("{$event}", true, true);
+element.dispatchEvent(event);
+}({{ELEMENT}}));
+JS;
```
Comment (Chekote) [inline src/Selenium2Driver.php:296]: Please match the indentation use in `dragTo()`:

```php
        $script = <<<JS
(function (element) {
    var event = document.createEvent("HTMLEvents");
    event.initEvent("{$event}", true, true);
    event.dataTransfer = {};
    element.dispatchEvent(event);
}({{ELEMENT}}));
JS;
```

File: src/Selenium2Driver.php [hunk 2]
```diff
@@ -278,6 +278,24 @@ private function executeJsOnElement(Element $element, $script, $sync = true)
...
+        $this->withSyn()->executeJsOnXpath($xpath, $script);
+    }
```
Comment (Chekote) [inline src/Selenium2Driver.php:298]: Please replace the similar code in `dragTo()` with calls to this new method:

## Before

```php
    public function dragTo($sourceXpath, $destinationXpath)
    {
        $source      = $this->findElement($sourceXpath);
        $destination = $this->findElement($destinationXpath);

        $this->wdSession->moveto(array(
            'element' => $source->getID()
        ));

        $script = <<<JS
(function (element) {
    var event = document.createEvent("HTMLEvents");
    event.initEvent("dragstart", true, true);
    event.dataTransfer = {};
    element.dispatchEvent(event);
}({{ELEMENT}}));
JS;
        $this->withSyn()->executeJsOnElement($source, $script);

        $this->wdSession->buttondown();
        $this->wdSession->moveto(array(
            'element' => $destination->getID()
        ));
        $this->wdSession->buttonup();

        $script = <<<JS
(function (element) {
    var event = document.createEvent("HTMLEvents");
    event.initEvent("drop", true, true);
    event.dataTransfer = {};
    element.dispatchEvent(event);
}({{ELEMENT}}));
JS;
        $this->withSyn()->executeJsOnElement($destination, $script);
    }
```

## After

```php
    public function dragTo($sourceXpath, $destinationXpath)
    {
        $destination = $this->findElement($destinationXpath);

        $this->wdSession->moveto(array(
            'element' => $source->getID()
        ));

        $this->dispatchJSEventOnElement('dragstart', $sourceXpath);
        
        $this->wdSession->buttondown();
        $this->wdSession->moveto(array(
            'element' => $destination->getID()
        ));
        $this->wdSession->buttonup();

        $this->dispatchJSEventOnElement('drop', $destinationXpath);
    }
```

**Review by Chekote — CHANGES_REQUESTED** [review]
Please also add the `moveToAll` method that we discussed. Thanks. 🙏

File: src/Selenium2Driver.php (hunk 2 above)
Comment (rashidsaf) [reply src/Selenium2Driver.php:298]: I knew, i forget something..

Comment (Chekote) [reply src/Selenium2Driver.php:298]: Good idea 👍

**Review by Chekote — APPROVED** [review]

**Pull request merged**